### PR TITLE
Adding support to terratest

### PR DESCRIPTION
**Pull request by danielcrisap**
In this PR we are adding support to [Terratest](https://github.com/gruntwork-io/terratest) and implement CI pipeline.

 ----

Covered modules:
-> **iam_account**
-> **iam_assumeble_role**
-> **iam_group_complete**
-> **iam_group_with_policies**
-> **iam_user_test**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            }
+        }
+        stage('Terraform Init') {
+            steps {
+                sh 'find . -type f -name "*.tf" -exec dirname {} \\;|sort -u | while read m; do (cd "$m" && terraform init -input=false -backend=false) || exit 1; done'
+            }
+        }
+        stage('Terraform Validate') {
+            steps {
+                sh 'find . -name ".terraform" -prune -o -type f -name "*.tf" -exec dirname {} \\;|sort -u | while read m; do (cd "$m" && terraform validate && echo "√ $m") || exit 1 ; done'
```
Comment (wmaramos) [inline Jenkinsfile:34]: I think would be interesting put these commands in Makefile, it'll be easier to us run the same steps that the CI.

File: test/go.sum [hunk 2]
```diff
@@ -0,0 +1,518 @@
+cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
```
Comment (wmaramos) [inline test/go.sum:1]: Is ist file auto generate? I'd say Is it something like Gemfile.lock?

File: Jenkinsfile (hunk 1 above)
Comment (danielcrisap) [reply Jenkinsfile]: Improve done in `547afc5`

File: test/go.sum (hunk 2 above)
Comment (danielcrisap) [reply test/go.sum:1]: Yes, it is generated automatically, based on `go.mod`

**Pull request merged**